python/lsst/pipe/tasks/postprocess.py

```diff
@@ -98,6 +133,26 @@ def runDataRef(self, patchRefList):
         mergedCatalog = self.run(catalogs, tract=dataId['tract'], patch=dataId['patch'])
         self.write(patchRefList[0], mergedCatalog)
 
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+
+        measDict = {ref.dataId['band']: {'meas': cat} for ref, cat in
+                    zip(inputRefs.inputCatalogMeas, inputs['inputCatalogMeas'])}
+        forcedSourceDict = {ref.dataId['band']: {'forced_src': cat} for ref, cat in
+                            zip(inputRefs.inputCatalogForcedSrc, inputs['inputCatalogForcedSrc'])}
+
+        catalogs = {}
+        for band, catDict in measDict.items():
+            catalogs[band] = {}
+            catalogs[band]['meas'] = measDict[band]['meas']
+            catalogs[band]['forced_src'] = forcedSourceDict[band]['forced_src']
+            catalogs[band]['ref'] = inputs['inputCatalogRef']
+
+        dataId = butlerQC.quantum.dataId
+        parq = self.run(catalogs=catalogs, tract=dataId['tract'], patch=dataId['patch'])
+        outputs = pipeBase.Struct(outputCatalog=parq.toDataFrame())
```

Comment:
I don't know if there's a big overhead to converting to and from parquet and dataFrames, but I think it's odd that the last thing the `run` method does is convert the dataframe to parquet, and the first thing this code does is convert it back to a dataframe.  I know that's what the gen2 code expects, but in terms of planning for gen2 removal, I think that `run` should return a dataframe and `runDataRef` could then parquet-ify the table?

Reply:
I agree.  ParquetTable will disappear with along with Gen2.  
Was avoiding API changes on this one, but if the reviewer advocates for API changes, well then I guess I have to :)